### PR TITLE
Fix a whole lot of conditional assignments on confirm & thank you pages

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fix a whole lot of conditional assignments on confirm & thank you pages

Before
----------------------------------------
image

After
----------------------------------------
All assigned unconditionally

image

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contribute/Form/Contribution/ThankYou.php [hunk 1]
```diff
@@ -51,8 +51,9 @@ public function preProcess(): void {
     $this->assign('initial_reminder_day', CRM_Utils_Array::value('initial_reminder_day', $this->_values));
     // Link (button) for users to create their own Personal Campaign page
     if ($linkText = CRM_PCP_BAO_PCP::getPcpBlockStatus($this->getContributionPageID(), 'contribute')) {
+      $contributionPageID = $this->getContributionPageID();
       $linkTextUrl = CRM_Utils_System::url('civicrm/contribute/campaign',
-        "action=add&reset=1&pageId=' . $this->getContributionPageID() . '&component=contribute",
+        "action=add&reset=1&pageId=' . $contributionPageID . '&component=contribute",
```
Comment (mattwire) [inline CRM/Contribute/Form/Contribution/ThankYou.php:56]: Is this change necessary?

File: CRM/Contribute/Form/ContributionBase.php [hunk 2]
```diff
@@ -710,55 +710,25 @@ public function setDefaultValues() {
...
+    $this->assign('amount', $this->getMainContributionAmount() > 0 ? CRM_Utils_Money::format($this->getMainContributionAmount(), NULL, NULL, TRUE) : NULL);
+
+    $isRecurEnabled = isset($this->_values['is_recur']) && !empty($this->_paymentProcessor['is_recur']);
+    $this->assign('is_recur_enabled', $isRecurEnabled);
+    $this->assign('is_recur', $isRecurEnabled ? ($this->_params['is_recur'] ?? NULL) : NULL);
+    $this->assign('frequency_interval', $isRecurEnabled ? ($this->_params['frequency_interval'] ?? NULL) : NULL);
+    $this->assign('frequency_unit', $isRecurEnabled ? ($this->_params['frequency_unit'] ?? NULL) : NULL);
+    $this->assign('installments', $isRecurEnabled ? ($this->_params['installments'] ?? NULL) : NULL);
+    $isPledgeEnabled = CRM_Core_Component::isEnabled('CiviPledge') && !empty($this->_params['is_pledge']);
+    // TODO: Assigned variable appears to be unused
```
Comment (mattwire) [inline CRM/Contribute/Form/ContributionBase.php:726]: You mean `pledge_enabled` is unused or all of the below?

File: CRM/Contribute/Form/Contribution/ThankYou.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Contribute/Form/Contribution/ThankYou.php:56]: @mattwire something weird was going on - it was giving me errors & no matter how I stared it didn't go away until I did this

File: CRM/Contribute/Form/ContributionBase.php (hunk 2 above)
Comment (eileenmcnaughton) [reply CRM/Contribute/Form/ContributionBase.php:726]: yeah - that comment goes with pledge enabled - it's moved, not new

Comment (eileenmcnaughton) [reply CRM/Contribute/Form/ContributionBase.php]: I updated the comment

**Comment by eileenmcnaughton** [comment]
test this please

File: CRM/Contribute/Form/Contribution/ThankYou.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Contribute/Form/Contribution/ThankYou.php:56]: OK - sleep helped https://github.com/civicrm/civicrm-core/pull/28374
